dnfdaemon-server/session.hpp

```diff
@@ -73,6 +77,26 @@ class Session {
     void fill_sack();
     bool read_all_repos();
 
+    template <class S>
+    void run_in_thread(S & service, sdbus::MethodReply (S::*method)(sdbus::MethodCall &&), sdbus::MethodCall && call) {
+        auto worker = std::thread(
+            [&method, &service, this](sdbus::MethodCall call) {
+                try {
+                    auto reply = (service.*method)(std::move(call));
+                    reply.send();
+                } catch (std::exception & ex) {
+                    auto reply = call.createErrorReply({dnfdaemon::ERROR, ex.what()});
+                    try {
+                        reply.send();
+                    } catch (...) {
+                    }
```

Comment:
Yes. The `MethodReply::send()` method from the sdbus-c++ library can generate an `sdbus::Error` exception.
Silent discarding of all errors can make it difficult to find out why something isn't working. What about writing errors into the log?
Write here at least TODO.

Reply:
Right. I've added a fixup commit with logging the error. If you are OK with it I will squash the fixup.